daemon/snapshotter/mount.go

```diff
@@ -93,7 +94,14 @@ func (m *refCountMounter) Mount(mounts []mount.Mount, containerID string) (targe
 	}()
 
 	root := m.idMap.RootPair()
-	if err := idtools.MkdirAllAndChown(target, 0o700, root); err != nil {
+	dirID := idtools.Identity{
+		UID: idtools.CurrentIdentity().UID,
+		GID: root.GID,
+	}
+	if err := idtools.MkdirAllAndChown(path.Dir(target), 0o710, dirID); err != nil {
+		return "", err
+	}
```

Comment:
Can you change this to use `filepath.Dir()` ?

I also thought for a long time that `path` is for `/` (and `filepath` only if you need cross-platform); https://github.com/golang/go/blob/2e7e1d2e8d96ddf2aff00feb645cfa94bc9e4786/src/path/path.go#L5-L6

> Package path implements utility routines for manipulating slash-separated paths.

But a second paragraph was added at some point! https://github.com/golang/go/commit/7465bfb1ea1a7ddfec9b267587ee9e6200514f3f

> To manipulate operating system paths, use the path/filepath package.

While at it, perhaps also inline that `dirID` var (I see the other implementations also have the above issues, but we can look at those separately)

```suggestion
	if err := idtools.MkdirAllAndChown(filepath.Dir(target), 0o710, idtools.Identity{
		UID: idtools.CurrentIdentity().UID,
		GID: root.GID,
	}); err != nil {
		return "", err
	}
```